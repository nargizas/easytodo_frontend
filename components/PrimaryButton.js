// import { useLinkProps } from '@react-navigation/native';
import { View, Text, Pressable, StyleSheet } from 'react-native';
// import { useNavigation } from '@react-navigation/native'

function PrimaryButton({ children, onPress }) {
    return (
        <View style={styles.buttonOuterContainer}>
            <Pressable style={styles.buttonInnerContainer} onPress={onPress} android_ripple={{ color: '#c7c7c7' }}>
                <Text style={styles.buttonText}>{children}</Text>
            </Pressable>
        </View>
    );
}

export default PrimaryButton;

const styles = StyleSheet.create({
    buttonOuterContainer: {
        borderRadius: 36,
        marginVertical: 24,
        marginHorizontal: 16,
        overflow: 'hidden',
    },
    buttonInnerContainer: {
        backgroundColor: '#f5f5f5',
        paddingVertical: 8,
        paddingHorizontal: 16,
        elevation: 2,

    },
    buttonText: {
        color: 'black',
        fontSize: 18,
        opacity: 54,
        textAlign: 'center'
    }
})