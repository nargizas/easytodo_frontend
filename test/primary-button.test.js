import React from 'react';
import renderer from 'react-test-renderer';

import PrimaryButton from '../components/PrimaryButton';

it('renders correctly', () => {
  const tree = renderer.create(<PrimaryButton />).toJSON();
  expect(tree).toMatchSnapshot();
});
