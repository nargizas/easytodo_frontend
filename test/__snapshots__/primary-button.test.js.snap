// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    {
      "borderRadius": 36,
      "marginHorizontal": 16,
      "marginVertical": 8,
      "overflow": "hidden",
    }
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "#f5f5f5",
        "elevation": 2,
        "paddingHorizontal": 16,
        "paddingVertical": 8,
      }
    }
  >
    <Text
      style={
        {
          "color": "black",
          "fontSize": 18,
          "opacity": 54,
          "textAlign": "center",
        }
      }
    />
  </View>
</View>
`;
